docs/guides/es2k/compiling-p4-programs.md

```diff
@@ -0,0 +1,117 @@
+# Compiling P4 Programs for ES2K
```

Comment:
In all the quoted documentations below, use ```bash for all commands that can be copy pasted. Avoid showing the command prompt as it will mess up with copy pasting commands. Also, avoid putting # comments inside the quotes.